### ISSUE TITLE
Roadmap for Chaiscript?

### ISSUE THREAD

**Issue by patlecat** [ChaiScript/ChaiScript#498]
Hi, I'd love to know when new updates are planned and which ones. Especially documentation and tool integration is dearly missing I think.

Do you have a roadmap maybe so we know what to expect? Or have you abandoned the project?

**Comment by germandiagogomez** [comment]
I think this project is worth a successor in the case it is abandoned. I never found a project that is so easy to integrate with C++. But I miss a few items (in which I could work myself as I keep using it in my own projects). I'd rather contribute to this project and not having to fork, but it seems quite abandoned. Areas that I would improve immediately:

1.  documentation. With good documentation there are a few things that must be reverse-engineered but that are already doable actually.
2. some kind of coroutine integration. This is the biggest pain point for me compared to the alternatives I would have chosen otherwise (sol2 for Lua or squirrel language). But the ease of integration won.
3. I would like to optimize attribute access. I think that having non-immutable strings for this (std::map<std::string, Boxed_Value> is quite wasteful.

4. (Idea): be able to embed Chaiscript code as C++ constexpr and do some optimizations. For my use case it goes like this: I work with a mix of C++/Chaiscript to have the convenience of interactive/scripted development. Once I have it, I embed the Chaiscript code. It would be nice if that code could be optimized in the final compilation and embedded as a resource.

**Comment by patlecat** [comment]
Yeah I would hate to see it fade away. The foundation is very good and the orientation on performance and easy integration is key! It just lacks in a few other areas where help from others would be beneficial. Sadly I have no clue how to write a language I would need a mentor that has time to help. Maybe others feel the same way?

**Comment by ninnghazad** [comment]
I don't think it will fade away - it's just too good.
Although i feel it rather lacks performance, working with its way of integration is just sweet.

Thought about supporting the founder's patreon, but i am not interested in his other work, as much as i am interested in/need ChaiScript - and my two coins probably wouldn't matter much in the way of enabling to spend more time on ChaiScript.

For me ChaiScript isn't really missing any features i would need, other than more speed.

Coroutines i think would be a pain to add - it might be worth waiting for [coroutines in c++](https://en.cppreference.com/w/cpp/language/coroutines) and we might be able to just use them with ChaiScript as is.

I don't know if a bytecode system would be nice - it might not be any faster but be a lot of work.

More optimizers i think would be nice. When sniffing around the source i imagined things my naive mind would call "block-optimizers" or something like that. That would aggregate common structures in the tree into one. Like for-if, if-if-if or something like that in order to reduce the enormous number of calls a large script produces.
Tracing ChaiScript parts of my code produces trees as big as the sky - i imagine reducing the number of small calls by aggregating would be beneficial.

My case of use is game-dev.

**Comment by germandiagogomez** [comment]
@lefticus if I could get a place where to put some documentation about how to use ChaiScript with more examples it would be a nice pointer. I could send a PR as I keep incrementally adding things. Since I have been using ChaiScript for a while, I think it could be valuable to others to add at least some more documentation beyond the cheat sheet.

**Comment by RobLoach** [comment]
Lefticus has given me commit access on ChaiScript, so if he's not available to merge Pull Requests, I could do so. Thanks all!

One thing I would love to see is getting Appveyor to pass.

**Comment by germandiagogomez** [comment]
@RobLoach I have #499 pending there. If you want to take a look... it is quite simple and it is about integration of json with the native object model. 

Feedback is welcome on how to proceed if the tiny improvement is welcome. Also, alternative implementations, feedback, suggestions or whatever.

**Comment by lefticus** [comment]
closing issue, I'm actively trying to catch up on things right now.
